src/components/input/SelectNext.tsx

```diff
@@ -90,13 +123,17 @@ function SelectMain<T>({
   elementRef,
 }: SelectProps<T>) {
   const [listboxOpen, setListboxOpen] = useState(false);
-  const [shouldListboxDropUp, setShouldListboxDropUp] = useState(false);
   const closeListbox = useCallback(() => setListboxOpen(false), []);
   const wrapperRef = useRef<HTMLDivElement | null>(null);
   const listboxRef = useRef<HTMLDivElement | null>(null);
   const listboxId = useId();
   const buttonRef = useSyncedRef(elementRef);
   const buttonId = useId();
+  const shouldListboxDropUp = useShouldDropUp(
+    buttonRef.current ?? null,
+    listboxRef.current,
+    listboxOpen,
+  );
```

Comment:
You need to pass in the refs here, rather than the current value. If, for any reason, `listboxOpen` was true on the first render, the layout effect wouldn't get called.

Reply:
Ugh, you are right